### PR TITLE
Fix bug pertaining to timestamp override

### PR DESCRIPTION
**Pull request by CharlyF**
### What does this PR do?

The core change here is removing  `now = metricTimestamp`

The timestamp of the datapoint of a metric **should not** be used as reference to compute whether we are allowed to scale.
What will happen is if the metric is 90s old (which is within range of acceptability) comparing the LastScalingTime to it and to the forbidden windows gives an incorrect result.
See the example `Forbidden window uses the right timestamp` in the tests.

Also fixed the tests, without the Reactor (get specifically) we would get empty objects, so everything would pass because it would be comparing nil objects.

### Motivation

Fix bug.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: controllers/watermarkpodautoscaler_controller_test.go [hunk 1]
```diff
@@ -307,37 +304,49 @@ func newRequest(ns, name string) reconcile.Request {
...
-	return nil, time.Time{}, nil
-}
-
-//
-//// GetObjectMetric gets the given metric (and an associated timestamp) for the given
-//// object in the given namespace
-func (f fakeMetricsClient) GetObjectMetric(metricName string, namespace string, objectRef *v2beta2.CrossVersionObjectReference, metricSelector labels.Selector) (int64, time.Time, error) {
-	return 0, time.Time{}, nil
+func addUpdateReactor(s *fakescale.FakeScaleClient) {
+	s.AddReactor("update", "deployments", func(rawAction testcore.Action) (handled bool, ret runtime.Object, err error) {
```
Comment (vboulineau) [inline controllers/watermarkpodautoscaler_controller_test.go:308]: For consistency between `addUpdateReactor` and `addGetReactor`, either put all reactor functions inline (update) or split (get). I have no strong preference between the two.

**Review by vboulineau — APPROVED** [review]
